metrics/sources/summary/summary.go

```diff
@@ -194,10 +194,22 @@ func (this *summaryMetricsSource) decodePodStats(metrics map[string]*MetricSet,
 
 	for _, container := range pod.Containers {
 		key := PodContainerKey(ref.Namespace, ref.Name, container.Name)
+		if _, exist := metrics[key]; exist && containerIsTerminated(&container, metrics[key].CreateTime) {
```

Comment:
IIUC the logic is as follow:
- if we first encounter an old container, then the new one will just replace it
- if we first encounter the new, this check will ensure that the old will not replace it

It's not that obvious. Please add a comment explaining this.